src/listpack.c

```diff
@@ -913,6 +913,34 @@ unsigned char *lpFind(unsigned char *lp, unsigned char *p, unsigned char *s,
     return lpFindCbInternal(lp, p, &arg, lpFindCmp, skip);
 }
 
+struct lpFindIntegerArg {
+    long long target; /* Item to search */
+};
+
+/* Comparator function to find Integer item */
+static inline int lpFindIntegerCmp(const unsigned char *lp, unsigned char *p,
+                                   void *user, unsigned char *s, long long ll) {
+    (void) lp;
+    (void) p;
+    struct lpFindIntegerArg *arg = user;
+    if (s != NULL) return 1;
+    return (ll == arg->target) ? 0 : 1;
+}
+
+/* Search for a specific long long integer value in the listpack.
+ *
+ * 'lp' is the pointer to the listpack.
+ * 'p' is the starting position within the listpack (if NULL, search starts from the first entry).
+ * 'value' is the target integer value to search for.
+ * 'skip' specifies the number of entries to skip between each comparison.
+ *
+ * Returns a pointer to the matching entry if found, otherwise NULL. */
+unsigned char *lpFindInteger(unsigned char *lp, unsigned char *p,
+                             long long value, unsigned int skip) {
+    struct lpFindIntegerArg arg = { .target = value };
```

Comment:
please note that`target` is 32bit with 32bit platform, we can change as follows.
```suggestion
    struct lpFindIntegerArg arg = { .target = &value };
```

Reply:
please ignore above, we can directly pass &value instead lpFindIntegerArg.

Reply:
I agree, and I've updated the code based on your suggestion :)